File: redsheet-portal/src/app/database-models/target.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { ProjectNegotiation,  vTargetRecord  } from './'
export class Target {
    constructor(target?: Target  | vTargetRecord )  {
        let properties = [ "targetDetail", "targetId", "projectNegotiationId",];
        for (let property of properties)
            if (target && target[property])
                this[property] = target[property];
    }
 
    @required()
    @maxLength(200)
	targetDetail : string =   undefined;
 
	targetId : number =   0 ;
 
    @range(0,2147483647)
	projectNegotiationId : number =   undefined;
	projectNegotiation : ProjectNegotiation  ;


}

File: RedSheet.DbEntities/Models/Target.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("Targets", Schema = "dbo")]
	[Serializable]
	public partial class Target  
	{
 
	[Required] 
	[MaxLength(200)]
	public string TargetDetail { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int TargetId { get; set; } 
 
	[RelationshipTableAttribue("ProjectNegotiations","dbo")]
	//Relationships
	public int ProjectNegotiationId { get; set; } 
 
	[ForeignKey("ProjectNegotiationId")]
	public ProjectNegotiation ProjectNegotiation { get; set; } 

	public Target()
	{
    }

    }
}

